### PR TITLE
SWC-6465: Virtual Table Support

### PR DESCRIPTION
**Pull request by nickgros**
- Add support for creating VirtualTables in experimental mode
- Disable versioning UI for SubmissionView (SWC-6423)

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java [hunk 1]
```diff
@@ -1626,12 +1626,29 @@ private void configureShareAction() {
...
+  public static boolean isVersionSupported(Entity entity) {
+    // The following table types inherit Versionable, but do not actually support snapshot versioning
     boolean isMaterializedView = entity instanceof MaterializedView;
-    return entity instanceof Versionable && !isMaterializedView;
+    boolean isVirtualTable = entity instanceof VirtualTable;
+    boolean isSubmissionView = entity instanceof SubmissionView;
+    boolean isNonSnapshottableTable =
+      isMaterializedView || isVirtualTable || isSubmissionView;
+    return entity instanceof Versionable && !isNonSnapshottableTable;
+  }
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java:1637]: Fixes SWC-6423

File: src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java [hunk 2]
```diff
@@ -1626,12 +1626,29 @@ private void configureShareAction() {
...
+
+  public static boolean isDefinedBySql(Entity entity) {
+    return entity instanceof HasDefiningSql;
+  }
+
+  public static boolean isEditCellValuesSupported(Entity entity) {
+    // EntityRefCollectionView results are not editable, even if the user has permissions, because rows may reference immutable versions (SWC-5870, SWC-5903)
+    return QueryResultEditorWidget.isTableTypeQueryResultEditable(
+      entity.getClass()
+    );
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java:1651]: QueryResultEditorWidget also defines this function, so use it here and have a single source of truth

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/QueryResultEditorWidget.java [hunk 3]
```diff
@@ -367,11 +369,12 @@ private void doHideEditor() {
...
-      tableType.equals(TableType.dataset) ||
-      tableType.equals(TableType.dataset_collection) ||
-      tableType.equals(TableType.materialized_view)
-    );
+    return isTableTypeQueryResultEditable(tableType.getClazz());
+  }
+
+  public static boolean isTableTypeQueryResultEditable(Class<?> clazz) {
+    // EntityRefCollectionView results are not editable, even if the user has permissions, because rows may reference immutable versions (SWC-5870, SWC-5903)
+    return clazz.equals(TableEntity.class) || clazz.equals(EntityView.class);
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/QueryResultEditorWidget.java:377]: Easier to list those table types that can be edited...notably, the result changes for `SubmissionView` -- I don't think this was functional in that case.

File: pom.xml [hunk 4]
```diff
@@ -19,7 +19,7 @@
     <webappDirectory>
       ${project.build.directory}/${project.build.finalName}
     </webappDirectory>
-    <synapse.version>446.0</synapse.version>
+    <synapse.version>457.0</synapse.version>
```
Comment (jay-hodgson) [inline pom.xml:22]: It's been a while!

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**